#! /bin/sh /usr/share/dpatch/dpatch-run
## 04_fix_resume_pause.dpatch by dodamn <dodamn@gmail.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: When unpausing is on, resume does not work. This patch fix it.

@DPATCH@

diff --git a/alsa.c b/alsa.c
index 64e12f7..e68d718 100644
--- a/alsa.c
+++ b/alsa.c
@@ -450,7 +450,9 @@ static int op_alsa_pause(void)
 		} else {
 			d_print("error: state is not RUNNING or PREPARED\n");
 		}
-	}
+	} else {
+        snd_pcm_drop(alsa_handle);
+    }
 	return 0;
 }
 
@@ -475,7 +477,9 @@ static int op_alsa_unpause(void)
 		} else {
 			d_print("error: state is not PAUSED nor PREPARED\n");
 		}
-	}
+	} else {
+        snd_pcm_prepare(alsa_handle);
+    }
 	return 0;
 }
 
